lib/model/model.go

```diff
@@ -1407,7 +1408,14 @@ func (m *Model) Request(deviceID protocol.DeviceID, folder, name string, size in
 		limiter.take(int(size))
 	}
 
-	res := NewRequestResult(int(size), limiter)
+	// The requestResult releases the bytes to the limiter when its Done method is called.
+	res := newRequestResult(int(size), limiter)
+	defer func() {
+		// Done it ourselves if it isn't returned due to an error
+		if err != nil {
+			res.Done()
+		}
```

Comment:
I'd make done/close be a noop if called multiple times, as tracking who called it where is hard.

Reply:
Done - doesn't really influence this part though (?).